WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift

```diff
@@ -248,6 +261,27 @@ final class DashboardViewModel {
         viewModel?.$showWebViewSheet.assign(to: &self.$showWebViewSheet)
         announcementViewModel = viewModel
     }
+
+    /// Sets up observer to decide store onboarding task lists visibility
+    ///
+    private func setupObserverForShowOnboarding() {
+        guard featureFlagService.isFeatureFlagEnabled(.dashboardOnboarding) else {
+            return
+        }
+
+        guard let storeOnboardingViewModel else {
+            showOnboardingSubscription = userDefaults.publisher(for: \.completedAllStoreOnboardingTasks)
+                .sink { [weak self] in
+                    self?.showOnboarding = !$0
+                }
+            return
+        }
```

Comment:
as the last refactoring suggestion, perhaps `storeOnboardingViewModel` can be either a lazy var or a `let` that's initialized with the `DashboardViewModel` to avoid having the userdefaults check in multiple places?